integration-tests/tests/src/tests/sync/mongo-db-client.ts

```diff
@@ -93,6 +93,438 @@ describe.skipIf(environment.missingServer, "MongoDB Client", function () {
+
+        const result = await collection.deleteOne();
+        expect(result.deletedCount).to.equal(1);
+      });
+
+      it("does not delete any document when there are no matches", async function (this: AppContext & Mocha.Context) {
+        await insertThreeDocuments();
+
+        const result = await collection.deleteOne({ _id: nonExistentId });
+        expect(result.deletedCount).to.equal(0);
+      });
+    });
+
+    describe("#deleteMany", function () {
+      it("deletes all documents using empty filter", async function (this: AppContext & Mocha.Context) {
+        await insertThreeDocuments();
+
+        const result = await collection.deleteMany();
+        expect(result.deletedCount).to.equal(3);
+      });
+
+      it("deletes documents using query selector", async function (this: AppContext & Mocha.Context) {
+        await insertThreeDocuments();
+
+        const result = await collection.deleteMany({ _id: { $gt: insertedId1 } });
+        expect(result.deletedCount).to.equal(2);
+      });
+
+      it("does not delete any document when there are no matches", async function (this: AppContext & Mocha.Context) {
+        await insertThreeDocuments();
+
+        const result = await collection.deleteMany({ _id: nonExistentId });
+        expect(result.deletedCount).to.equal(0);
+      });
+    });
+
+    describe("#count", function () {
+      it("returns total number of documents using empty filter", async function (this: AppContext & Mocha.Context) {
+        const countBefore = await collection.count();
+        expect(countBefore).to.equal(0);
+
+        await insertThreeDocuments();
+
+        const countAfter = await collection.count();
+        expect(countAfter).to.equal(3);
+      });
+
+      it("returns number of documents using query selector", async function (this: AppContext & Mocha.Context) {
+        await insertThreeDocuments();
+
+        const count = await collection.count({ _id: { $gt: insertedId1 } });
+        expect(count).to.equal(2);
+      });
+
+      it("returns zero when there are no matches", async function (this: AppContext & Mocha.Context) {
+        const count = await collection.count({ _id: nonExistentId });
+        expect(count).to.equal(0);
+      });
+    });
+
+    describe("#aggregate", function () {
+      it("aggregates documents using multiple pipeline stages", async function (this: AppContext & Mocha.Context) {
+        await insertThreeDocuments();
+
+        const result = await collection.aggregate([
+          // Filter all docs with `_id` > `insertedId1`.
+          { $match: { _id: { $gt: insertedId1 } } },
+          // Return a single doc (`_id: null`) with property `count` set to the number of filtered docs.
+          { $group: { _id: null, count: { $sum: 1 } } },
+          // Remove the `_id` field from the result.
+          { $project: { _id: false } },
+        ]);
+        expect(result).to.deep.equal([{ count: 2 }]);
+
+        // Note:
```

Comment:
It is documented in `integration-tests/baas/README.md` but please consider to copy steps 1-4 into it.

Reply:
[Done](https://github.com/realm/realm-js/pull/5475) ✅